### ISSUE TITLE
LambdaRank relavance value type

### ISSUE THREAD

**Issue by Kiona0405** [dmlc/xgboost#5618]
In demo/rank/rank_sklearn.py file, I added the following code.
Then model training becomes wrong. 

```python
model = xgb.sklearn.XGBRanker(**params)

# added code.  y value becomes float type
y_train /= 3.0
y_valid /= 3.0
y_test /= 3.0
```

Followings are result of execution.
```
rank ➤ python rank_sklearn.py                                                                                                                           
[0]	eval_0-map:1.00000
[1]	eval_0-map:1.00000
[2]	eval_0-map:1.00000
[3]	eval_0-map:1.00000
```

It seems y value requires integer type(discrete).
But original Lambdarank paper doesn't require discreteness.
Is it as expected? If so, please write it at readme.

**Comment by hcho3** [comment]
>  original Lambdarank paper doesn't require discreteness.

Which paper are you referring to? In section 2 of [this paper](https://www.microsoft.com/en-us/research/wp-content/uploads/2016/02/lambdarank.pdf), the labels are either binary (0, 1) or multi-level integers (0, 1, 2, ...). In particular, both MAP and NDCG metrics assume discrete integer labels.
